### PR TITLE
Convert `html` task to ES2015

### LINKED ISSUE #21: Rewrite html task in ES6
Rewrite [html.js](https://github.com/cloudfour/core-gulp-tasks/blob/09c3cba268b1f216b5112f7eadfd05265385abe1/lib/html.js) in ES6. It's only one task but it is by far the biggest.

Make sure tests pass!

### PR DESCRIPTION
**Pull request by lyzadanger**
Fixes #21 

I opted to leave module stuff (import/export) alone until we have all of this in place, as we were running into some fun with `requireDir`.

Also adds `.eslintrc`, if that's OK.

/cc @mrgerardorodriguez

### CODE REVIEW

File: lib/html.js [hunk 1]
```diff
@@ -1,14 +1,12 @@
-'use strict';
```
Comment (erikjung) [inline lib/html.js]: ?

File: lib/html.js [hunk 2]
```diff
@@ -22,14 +20,9 @@ var defaults = {
...
-  var opts = merge({}, defaults, options);
-  var sharedData = opts.sharedData;
-  var dest = opts.dest;
-  var layoutDir = opts.layoutDir;
-  var plugins = opts.plugins;
-  var src = opts.src;
-  var templateExt = opts.templateExt;
+module.exports = function(gulp, options) {
+  let opts = merge({}, defaults, options);
+  let { sharedData, dest, layoutDir, plugins, src, templateExt } = opts;
```
Comment (erikjung) [inline lib/html.js]: Nice. We should use this more.

**Comment by erikjung** [comment]
:+1: 

A few things we might all want to discuss standardizing:
- `let` vs. `var` (do we want to use this when block scoping isn't needed?)
- `use strict` (I noticed the removal here)
- Object merging (which dependency and method is best? Is there a native way?)

**Comment by lyzadanger** [comment]
@erikjung IIRC ES6 is always `stict mode` meaning that `use strict` is redundant... see http://stackoverflow.com/questions/31685262/not-recommended-to-write-out-use-strict-with-es6

**Comment by erikjung** [comment]
>  ES6 is always stict mode meaning that use strict is redundant

@lyzadanger Interesting, I didn't know about those rules. It seems that the key here is: "Module code is always strict mode code."

Looking at the comments on that thread, I wonder if this script actually qualifies as "module code" since it's technically not an ES6 module (using `module.exports` instead of `export`).

**Comment by lyzadanger** [comment]
@erikjung One way we could try (I suppose I could do this) is to introduce a strict error into this code and see if `node` freaks out without the `use strict` directive.

**Comment by lyzadanger** [comment]
@erikjung Hmmm, perhaps we do need to retain `use strict`...

1. Introducing a strict error (undeclared variable reference) into that task does _not_ cause errors (implying it's not running in strict mode)
2. [This thread](http://stackoverflow.com/a/29285330) suggests there is enough ambiguity that we should retain it.

#til 

I'll put the `use strict` back.

**Comment by lyzadanger** [comment]
@erikjung Ready for re-review! I replaced `use strict` and decided to go with the more explicit `var` instead of `let` because I don't strictly need the latter.

**Comment by erikjung** [comment]
:+1:

**Pull request merged**